contracts/test/v0.8/foundry/dev/special/NoCancelVRFCoordinatorV2.t.sol

```diff
@@ -0,0 +1,78 @@
+pragma solidity ^0.8.0;
+
+import {BaseTest} from "../../BaseTest.t.sol";
+import {NoCancelVRFCoordinatorV2} from "../../../../../src/v0.8/dev/special/NoCancelVRFCoordinatorV2.sol";
+import {MockLinkToken as LinkToken} from "./MockLinkToken.sol";
+
+contract NoCancelVRFCoordinatorV2Setup is BaseTest {
+  NoCancelVRFCoordinatorV2 s_coordinator;
+
+  LinkToken LINK;
+
+  address internal LINK_ETH_FEED_ADDRESS = 0x2B4Ef6FAC5F6BE758508605bd87356776BfbfbA0;
+  address internal BLOCKHASH_STORE_ADDRESS = 0x032477835Bc60C777321e1e09F1Fe6cFB82e111C;
+
+  function setUp() public virtual override {
+    BaseTest.setUp();
+
+    LINK = new LinkToken();
+    s_coordinator = new NoCancelVRFCoordinatorV2(address(LINK), LINK_ETH_FEED_ADDRESS, BLOCKHASH_STORE_ADDRESS);
+  }
+}
+
+contract NoCancelVRFCoordinatorV2_cancelSubscription is NoCancelVRFCoordinatorV2Setup {
+  address internal subscriptionOwner = 0x054684DA2dCcf0e79E72DcAb1b5db42E71b66c8A;
+
+  event SubscriptionCanceled(uint64 indexed subId, address to, uint256 amount);
+  event SubscriptionCreated(uint64 indexed subId, address owner);
+  event SubscriptionFunded(uint64 indexed subId, uint256 oldBalance, uint256 newBalance);
+
+  function testCancelSubscription() public {
+    vm.expectRevert();
+    s_coordinator.cancelSubscription(1, address(0)); // args don't matter, always reverts
+  }
+
+  function testOwnerCancelSubscriptionInvalidSubId() public {
+    vm.expectRevert(NoCancelVRFCoordinatorV2.InvalidSubscription.selector);
+    s_coordinator.ownerCancelSubscription(1); // non-existent subscription
+  }
+
+  function testOwnerCancelSubscription() public {
+    // create a subscription, fund it, then cancel it.
+    // assert that the funds are sent to the contract owner address.
+
+    // switch caller to subscriptionOwner so that the subscription owner and the
+    // contract owner are different.
+    vm.stopPrank();
+    vm.prank(subscriptionOwner);
+    vm.expectEmit(true /* checkTopic1 */, false /* checkTopic2 */, false /* checkTopic3 */, true /* checkData */);
```

Comment:
nit: instead of checkTopic1 maybe this should refer to subID? as well as checkData for subscriptionOwner.

Reply:
Fixed in [3fb1d7d](https://github.com/smartcontractkit/chainlink/pull/7928/commits/3fb1d7d4ba93e705819fb7035c62a8a0404a16fc)